HM3CoreKill/HM3CoreKill/docs/INIArgs.md

```diff
@@ -0,0 +1,16 @@
+INI Arguments
+==============
+
+ * EnableCheats - disabled by game code. Can be activated via CheatEnabler or our Launcher.
+ * DisableScripts - disable loading .dll with code of scripts. All characters will stay in T-pose without any logic.
+ * DisableTextures - disable loading of some UI' textures (preload menu, brief, pause menu and some other places)
+ * FreeCamMode - REMOVED
+ * EnableScriptRTC - unknown how to use
+ * ErrorLog - unknown how to use
+ * ScriptDebug
+ * ScriptDll
+ * Window - start game in window mode (0;0 + StartUpperPos)
+ * StartUpperPos - offset of window in windowed mode; Usage: StartUpperPos=50,50
+ * RecordFile - set file to gameplay record
+ * PlayFile - set file to play recorded gameplay
+ * SaveRecordRuntime - save game play record into file (declared in RecordFile). NOT WORKING (CRASH)
```

Comment:
I've previously documented this stuff here
https://github.com/HHCHunter/Hitman-BloodMoney/blob/master/CONFIG-FILES/HitmanBloodMoney.ini

[APPLICATION]
	; DefaultScene, this is the first scene which the engine loads, In this case it has been set to the Boot Menu.
		; It can be set to any level as they have their own .gms file (Glacier Menu System)
	DefaultScene=HitmanBloodMoney.gms
	; Streamfile, this is the file from which the engine stream's audio data from.
	streamfile pc_eng.str
	; Controller/Keyboard Bindings File
	ConfigFile=HM3GPAD.cfg
	; EnableNewSceneFormat, not sure what this is used for, research required.
	; EnableNewSceneFormat
	; ReadMainIni, setting this to 1 will read the executable's pre-baked configuration, thus ignoring this file.
	ReadMainIni 0
	; DisableMenuTimers, unsure what this does.
	DisableMenuTimers
	; Windows Only Configuration, that represents the background color.
	BackColor=255,255,255
	; DisableOptions, this disables the options menu in-game.
	DisableOptions 0
	; BuildTag, unsure how this is used, if uncommented will crash.
	; BuildTag Release

[CONFIGURATION]
	; WantBlood 1 enables blood.
	WantBlood 1
	; NoBlood 1 disables blood, however if WantBlood is set to 1, this will be ignored.
	NoBlood 0
	; Enable Vibration
	VibrationMode 0
	[MOUSE]
		UseMouse 1
		MouseSpeed 0.400000

[GRAPHICS OPTIONS]
	; Name of render dll to use.
	DrawDll RenderD3D.dll
	; Windowed Mode
	Window 0
	; Upper left corner of window.
	StartUpperPos 0,0
	; Set game render resolution.
	Resolution 3840x2160
	; Set screen color depth/bpp.
	ColorDepth 32
	; Antialiasing, multiples of 2. Max is 8.
	Antialias 8
	; Start in a window (not fullscreen).
	Window 0
	[V-SYNC]
		; Enable V-Sync
		EnableVSync 1
		; Enable Triple Buffering
		EnableTripleBuffering 1
	[DEBUG RENDERING OPTIONS]
		;DebugVertexShader is used to debug vertex shaders, if un-coommented performance is kill.
		;DebugVertexShader
		;DebugPixelShader
		ExpandVertexBuffers
	[LEVEL OF DETAIL]
		; Level of Detail max is 3.
		LevelOfDetail 3
		DrawDistance 1.200000
	[TEXTURES]
		; Un-comment to enable, DisableTextures disables background textures on overlay menu's, such as pause and briefing screens.
		; DisableTextures
		TextureResolution 0
		; Disable Environmental Bump Maps
		DisableEMBM 0
	[TEXTURE FILTERING]
		; Anistropic filtering options, Max is 16.
		Anisotropy 16
		; Disable Trilinear Texture Filtering, this should be off if Anisotropy is enabled.
		DisableTrilinearFiltering 1
	[SHADOWS]
		; SelfShadows enables self-shadowing on 47.
		SelfShadows 1
		; Shadow detail is the amount of details are present in shadows.
		ShadowDetail 2
		; ShadowResolution only effects dynamic shadows, not pre-baked shadows.
		ShadowResolution 4096
		BlurShadows 1
		DisableDropShadowBlur 0
	[LIGHTING]
		; DisableLight, uncomment to use, will disable lighting on materials that have dynamic shadows
		; DisableLight
		DisplayLights
	[SHADERS]
		DisableShaderCache
	[POST PROCESSING]
		; Optimization by reducing precision of pixel shaders, versions 2.0 and 3.0.
		RenderUsePartialPrecisionShader 0
		PostFilterLOD 2
		ShaderQuality VeryHigh
	[FOG]
		FogEnabled 0
		Culling 0

[AUDIO]
	[SOUND ENGINE]
	; Name of sound dll to use.
	SoundDll soundengine.dll
	;EnableDirectSAound, this is for using DirectX Audio implementation
	EnableDirectSound 1
	;EnableDieselPower
	;EnableOpenAL
	[ENVIRONMENTAL AUDIO]
	; Environmental Audio Extension
	UseEAX 1
	[AUDIO BUFFER]
	; Number of Sound Buffers, Max is 64
	NumSoundBuffers 64

	[VOLUME CONTROLLERS]
	; Volume controls
	Audio_Volume_Sfx 25.000000
	Audio_Volume_Dialog 25.000000
	Audio_Volume_Music 20.000000
	Audio_Volume_Ambience 25.000000
	Audio_Volume_Movie 30.000000
	[DEBUG]
		PrintSoundMapping 1

[SCRIPT]
	; ScriptIfDll
	; ScriptDll
	ScriptDebug 1
	; Allows scripts to print to zDebug (may be disabled for this build).
	ScriptDebugPrint 1
	; DisableScripts, Disable loading .dll code for scripts. All characters will stay in T-pose without any logic.
	DisableScripts 0
	ScriptEnableSanityCheck 1
	; Unknown how to use
	EnableScriptRTC 1
	; RecordFile, Set file to gameplay record
	RecordFile 0
	; Save Gameplay recording into a file (Save location is declared in RecordFile). NOT WORKING (CRASH)
	SaveRecordRuntime 0
	; PlayFile, Set the file to play previously recorded gameplay.
	PlayFile - set file to play recorded gameplay

[DEBUG]
	EnableDebugOptions 1
	EnableConsole 1
	; Disabled by game code. Can be activated via CheatEnabler or our Launcher.
	EnableCheats 1 
	; DEBUG OPTIONS set in HM3GPAD.cfg
	enablecontrollerdebugkeys 1
	; DebugMenus, 1 Enables Debug Menu's to be available from the mainmenu.
	DebugMenus 0
	DisplayMemory 1
	DebugAnimPack 0
	; FreeCamMode, REMOVED
	FreeCamMode 0
	; ErrorLog, Unknown how to use
	ErrorLog 0

[Console Commands - DEBUG]
	ConsoleCmd show_debugtext 1
	ConsoleCmd show_debuggeoms 1
	ConsoleCmd show_debug 1
	ConsoleCmd ActorDebugInfo 1

[Console Commands]
	ConsoleCmd ShowVision 1
	ConsoleCmd ActorInfo 1
	ConsoleCmd ScriptInfo 1
	ConsoleCmd ScriptFlightRecorder 1
	ConsoleCmd ScriptFlightRecorderFile scriptlog.txt

Reply:
Yes, this project is open to pull requests.
And thank you for your documentation :) 
Did you test your configuration? (ConsoleCmd option is very interesting for me, I will check that later)

Reply:
Really glad you haven't given up on this project!

I hand wrote the whole .ini file in my Git to use when I re-install the game, so it's easy to configure to my liking :)

Some of the documentation of the INI I have written is from old official IO-I documentation that was leaked with an old build of Hitman: Contacts where they documented it themselves.
Other parts are from my testing :)

The way some of the debug options work through the use of key combos is very interesting.
a lot of them are buy "Z" options, but I have no idea how they work.
It requires more research, unfortunately, it's out of my technical abilities :(

Reply:
Good, I'll attach a link to your docs in future commits :) 
For the now I'm working on reversing of scripting (DLLs in Scriptcs folder) and trying to manipulate of all actors on the scene (but the scripts prevents all my actions).